lib/klp.pm

```diff
@@ -45,14 +46,23 @@ sub install_klp_product {
     if ($livepatch_repo) {
         zypper_ar("$utils::OPENQA_FTP_URL/$livepatch_repo", name => "repo-live-patching");
     }
-    else {
+    elsif (!is_sle_micro) {
         zypper_ar("http://download.suse.de/ibs/SUSE/Products/$lp_module/$version/$arch/product/", name => "kgraft-pool");
         zypper_ar("$release_override http://download.suse.de/ibs/SUSE/Updates/$lp_module/$version/$arch/update/", name => "kgraft-update");
     }
 
-    # install kgraft product
-    zypper_call("in -l -t product $lp_product", exitcode => [0, 102, 103]);
-    zypper_call("mr -e kgraft-update") unless $livepatch_repo;
+    # Enable live patching
+    if (is_sle_micro) {
+        assert_script_run 'cp /etc/zypp/zypp.conf /etc/zypp/zypp.conf.orig';
+        assert_script_run 'sed -i "/^multiversion =.*/c\\multiversion = provides:multiversion(kernel)" /etc/zypp/zypp.conf';
+        assert_script_run 'sed -i "/^multiversion\.kernels =.*/c\\multiversion.kernels = latest" /etc/zypp/zypp.conf';
+        assert_script_run 'echo "LIVEPATCH_KERNEL=\'always\'" >> /etc/sysconfig/livepatching';
+        assert_script_run 'diff /etc/zypp/zypp.conf.orig /etc/zypp/zypp.conf';
```

Comment:
I think this will fail now, no? I meant to check the return code in some additional check :)

Reply:
So I will  rather remove that line  and not continue further, it was for info development anyway.